examples/kubernetes.yml

```diff
@@ -43,6 +43,7 @@ spec:
             periodSeconds: 3
             successThreshold: 1
             timeoutSeconds: 10
+        args: ["--max_size=$(BAZEL_REMOTE_MAX_SIZE)"] # ensure we override the default --max_size argument from the container.
```

Comment:
What happens if `$BAZEL_REMOTE_MAX_SIZE` is not set? Would it be better to remove the `--max_size=*` flags from the `container_image`'s in BUILD.bazel ?

Reply:
> What happens if $BAZEL_REMOTE_MAX_SIZE is not set?

It's going to exit 1, as no value was given to that flag. 

> Would it be better to remove the --max_size=* flags from the container_image's in BUILD.bazel ?

It would be better remove it entirely from the container image, but that will change the default from 5 GiB to unlimited. It's rather easy to do, but I think it'd be best to reason about the default values as whole than just dropping the flag from there. 

I think the correct path is: 

- set the default max size to 5GiB in the code, so regardless of the deployment method it's always the same. 
- remove all default flags from the containers 
- update the README 
- log the config on start at `INFO` level (to spot misconfiguration easily) 

@mostynb if you don't see any problem with that, I'll be happy to make a PR. But I think we'd rather merge this PR straight away to avoid having others stumbling across the same problem, especially since it's a very minor change on the docs.

Reply:
I'm not sure that I like the idea of setting a default cache size. 5GiB is too small for a lot of workloads, and I think we should require that a value is specified.

How about we remove the `--max_size=*` command line flag from the `//:bazel-remote-image` and `//:bazel-remote-image-arm64` targets, and add it explicitly in the example kubernetes.yml and the docker example in README.md?